### PR TITLE
add metadata to route

### PR DESCRIPTION
**Pull request by iherem**
When I can't specify additional values and send them to custom handlers. Then I add the metadata into the constuctor to specify the value of what will be added to the custom handler.

Suppose I want to specify the locale of route
`routes.add({ name: 'home', pattern: '/home', page: 'home', metadata: { locale: 'en' } })`

The custom handlers receive the metadata
```
const handler = routes.getRequestHandler(app, ({ req, res, route, query }) => {
  const { metadata: { locale }, page } = route;
  renderWithCache(req, res, page, locale, query);
});
```

and the simple case if you want to specify some of route for cache just add the `{ cache: true }` in the metadata

reference issue  #127

### CODE REVIEW

**Comment by trotzig** [comment]
Naive question perhaps, but why can’t the locale be part of the pattern?

**Comment by iherem** [comment]
Actually, I used it with my silly case. I don't know how other people do.
for example: 
I have the base route /landing if i go straight to url /uk/landing or /ph/landing the content is changing depend on the route country.
but the metadata can be everything you want to send to the server

**Comment by trotzig** [comment]
> but the metadata can be everything you want to send to the server

What part is sending stuff to the server? I'm a little confused 🙂

File: dist/index.js [hunk 1]
```diff
@@ -0,0 +1,290 @@
+'use strict';
```
Comment (thomasdom) [inline dist/index.js]: Is this file intended to be in the repository? If not, you should remove it.

File: src/index.js [hunk 2]
```diff
@@ -115,12 +115,13 @@ class Routes {
 }
 
 class Route {
-  constructor ({name, pattern, page = name}) {
+  constructor ({name, pattern, page = name, metadata = {}}) {
```
Comment (thomasdom) [inline src/index.js]: You should call it `meta`, like in Vue Router (https://router.vuejs.org/guide/advanced/meta.html)

**Comment by thomasdom** [comment]
Also, you should update documentation to reflect these changes, and bump the library minor version according to https://semver.org/.
